### PR TITLE
Use importlib instead of deprecated pkg_resources. Reorganize setup

### PR DESCRIPTION
**Pull request by mcara**
JWST Regression tests:

https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1069/

### CODE REVIEW

**Review by zacharyburnett — APPROVED** [review]

**Comment by mcara** [comment]
@zacharyburnett Should I split this PR into two? This one cannot be merged (tests are failing in Python 3.12) until a new `gwcs` version is released.

**Comment by zacharyburnett** [comment]
> @zacharyburnett Should I split this PR into two? This one cannot be merged (tests are failing in Python 3.12) until a new `gwcs` version is released.

I don't think that's necessary; can we wait for the new `gwcs` version, and then revisit this?

**Comment by pllim** [comment]
What is the difference between Azure "Publish" vs Actions "Build source and wheel distribution"? Do you really need both?

File: tweakwcs/__init__.py [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
 __author__ = 'Mihai Cara'
 
-from pkg_resources import get_distribution, DistributionNotFound
-try:
-    __version__ = get_distribution(__name__).version
-except DistributionNotFound:
-    # package is not installed
-    __version__ = 'UNKNOWN'
+
+from importlib.metadata import version
```
Comment (pllim) [inline tweakwcs/__init__.py:11]: I don't think you need importlib at all. Try this:

```
try:
    from ._version import version as __version__
except ImportError:
    __version__ = ''
```

File: azure-templates.yml
Comment (pllim) [inline azure-templates.yml:1]: If you advertise that you support python >= 3.7, then you should test also 3.7 and 3.8. But really 3.7 is already EOL, so I think it is okay to do `python_requires='>=3.7'`. Still you should keep the 3.8 job.

File: setup.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def get_transforms_data():
         'Topic :: Software Development :: Libraries :: Python Modules',
         'Development Status :: 3 - Alpha',
     ],
-    python_requires='>=3.5',
+    python_requires='>=3.7',
```
Comment (pllim) [inline setup.py:113]:
```suggestion
    python_requires='>=3.8',
```

**Comment by mcara** [comment]
According to mathematical induction.... if it works for 3.10 and it works for 3.9, it will work for 3.8. ... I think

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,130 @@
...
+    'README.rst',
+    'LICENSE.txt',
+    'CHANGELOG.rst',
+    '*.fits',
+    '*.txt',
+    '*.inc',
+    '*.cfg',
+    '*.csv',
+    '*.yaml',
+    '*.json',
```
Comment (pllim) [inline pyproject.toml:90]: Should be more specific on these ones. You don't want to bundle accidental data.

**Comment by pllim** [comment]
> if it works for 3.10 and it works for 3.9, it will work for 3.8

That is not a given. For example, you might be locked to an older numpy in Python 3.8, and then some code might not work in that combo if you stop testing it.

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,130 @@
...
+license-files = ["LICENSE.rst"]
+
+[tool.setuptools.packages.find]
+namespaces = false
+
+[tool.setuptools.package-data]
+"*" = [
+    'README.rst',
+    'LICENSE.txt',
+    'CHANGELOG.rst',
```
[the remark this answers is not in the document]
Comment (mcara) [reply pyproject.toml:81]: yes they are if I want them to be. One does not need to parse them in the code. They can be parsed with eyes.

[the remark this answers is not in the document]
Comment (pllim) [reply pyproject.toml:81]: You can include them in MANIFEST without making them actual package data for eye-feasting.

[the remark this answers is not in the document]
Comment (mcara) [reply pyproject.toml:81]: It's OK. `astropy` adds CITATION: https://github.com/astropy/astropy/blob/164a67784e669b30c859db904668e04b9f6e1bfa/pyproject.toml#L145 No one cares.

File: tweakwcs/__init__.py (hunk 1 above)
Comment (zacharyburnett) [reply tweakwcs/__init__.py:11]: I would also include `ModuleNotFoundError`:
```python
try:
    from ._version import version as __version__
except (ImportError, ModuleNotFoundError):
    __version__ = ''```

**Comment by mcara** [comment]
> > if it works for 3.10 and it works for 3.9, it will work for 3.8
> 
> That is not a given. For example, you might be locked to an older numpy in Python 3.8, and then some code might not work in that combo if you stop testing it.

1. I was kidding about mathematical induction;
2. If, let's say 3.7 is no longer under development and the numpy version which runs with 3.7 is unlikely to change, how can the code suddenly stop working? The only way would be to use some cutting edge features from Python 3.8 that are not available in 3.7. I think that is also very very unlikely to happen.

File: pyproject.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (pllim) [reply pyproject.toml:81]: The CITATION one is some copy-and-paste artifact from astropy maybe? astropy does have a Python API for citation, so it is included there on purpose.

File: tweakwcs/__init__.py (hunk 1 above)
Comment (pllim) [reply tweakwcs/__init__.py:11]: No one should blindly copy-paste stuff but point taken and it doesn't hurt.

File: pyproject.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcara) [reply pyproject.toml:85]: > astropy does have a Python API for citation

I must say: WOW!

[the remark this answers is not in the document]
Comment (pllim) [reply pyproject.toml:85]: https://www.astropy.org/acknowledging.html

```python
import astropy
astropy.__citation__
```

**Pull request merged**